pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -755,6 +772,23 @@ func buildListener(opts buildListenerOpts) *xdsapi.Listener {
 	}
 }
 
+// Return trace sampling if set correctly, or default if not.
+func getTraceSampling() float64 {
+	if traceSamplingEnv == "" {
+		return traceSamplingDefault
+	}
+	f, err := strconv.ParseFloat(traceSamplingEnv, 64)
+	if err != nil {
+		log.Warnf("PILOT_TRACE_SAMPLING not set to a number: %v", traceSamplingEnv)
+		return traceSamplingDefault
+	}
+	if f < 0.0 || f > 100.0 {
+		log.Warnf("PILOT_TRACE_SAMPLING out of range: %v", f)
+		return traceSamplingDefault
+	}
+	return f
+}
```

Comment:
this function woudl go away if its part of mesh global config

Reply:
Not completely - still need 0..100 check. And I don't think it's a global - we may add it as an annotation to the service or pod ( using the node meta ), or as a proper API.

Reply:
Yes, but thats part of the validation check. And we should add annotations if and only if its a property of the service registry that needs to be enhanced. Tracing config (just like authentication/rbac/mixer etc.) is configuration. Not a registry property. Anyway, thats a separate design discussion. My point is that when this becomes an API, this code will go away as all validation code is in validation.go